source/common/http/conn_manager_utility.cc

```diff
@@ -50,9 +50,24 @@ Network::Address::InstanceConstSharedPtr ConnectionManagerUtility::mutateRequest
   // our peer to have already properly set XFF, etc.
   Network::Address::InstanceConstSharedPtr final_remote_address;
   bool single_xff_address;
+  auto xff_num_trusted_hops = config.xffNumTrustedHops();
   if (config.useRemoteAddress()) {
-    final_remote_address = connection.remoteAddress();
     single_xff_address = request_headers.ForwardedFor() == nullptr;
+    // If there are any trusted proxies in front of this Envoy instance (as indicated by
+    // the xff_num_trusted_hops configuration option), get the trusted client address
+    // from the XFF.
+    if (xff_num_trusted_hops > 0) {
+      auto ret = Utility::getLastAddressFromXFF(request_headers, xff_num_trusted_hops - 1);
```

Comment:
nit: I think you could just assign to final_remote_address directly  via `(...).address_`